src/ResourceManager/Network/Commands.Network/NetworkWatcher/SetAzureNetworkWatcherConfigFlowLogCommand.cs

```diff
@@ -91,13 +132,89 @@ public class SetAzureNetworkWatcherConfigFlowLogCommand : NetworkWatcherBaseCmdl
         [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
         public SwitchParameter AsJob { get; set; }
 
+        [Alias("EnableTA")]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Flag to enable/disable retention.",
+            ParameterSetName = SetFlowlogByResourceWithTAByResource)]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Flag to enable/disable retention.",
+            ParameterSetName = SetFlowlogByResourceWithTAByDetails)]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Flag to enable/disable retention.",
+            ParameterSetName = SetFlowlogByNameWithTAByResource)]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Flag to enable/disable retention.",
+            ParameterSetName = SetFlowlogByNameWithTAByDetails)]
+        [ValidateNotNull]
+        public SwitchParameter EnableTrafficAnalytics { get; set; }
```

Comment:
All that ValueFromPipelineByPropertyName does is enable the user to pass in the parameter by passing.  For example, if the user pipes in an object with property "ResourceGroupName" that is a string, the cmdlet will set "ResourceGroupName" to that property's value.  In the case of a switch parameter, this usually does not make sense.  It will not allow the user to pass in "-EnableTrafficAnalytics $true" - that will never be possible with a switch parameter.